### ISSUE TITLE
ERROR: GoogLeNet

### ISSUE THREAD

**Issue by KangKyungSoo** [NVIDIA/DIGITS#82]
TEST RESULT

![test](https://cloud.githubusercontent.com/assets/11133669/7229487/29bb1aa2-e79f-11e4-9658-e8ecb75733fc.png)

TRAIN RESULT

![train](https://cloud.githubusercontent.com/assets/11133669/7229509/7b11be88-e79f-11e4-84cc-5549304536e5.png)

**Comment by lukeyeager** [comment]
@KangKyungSoo, please ask your question at our [digits-users](https://groups.google.com/d/forum/digits-users) google group, and please give us some more information on what your problem is. It looks like your model hasn't learned anything, but you're not giving us much to work with.

**Comment by KangKyungSoo** [comment]
If you really want to know in detail on above ERROR sitation, please let me know.

**Comment by lukeyeager** [comment]
Of course I do, I'd just prefer to do it elsewhere. I suspect it's a matter of formatting the data improperly or something. But I definitely want to know if there's a bug in DIGITS or if there's a way to clarify the instructions so that people can avoid your problem in the future.

**Comment by KangKyungSoo** [comment]
My data is made with the ImageNet data, and it worked well with AlexNet. 
And there was no problem in traing with GoogLeNet(7 days) as you can see at above TRAINING image.
But the test results was that nice as you can see at above TEST image. 
If you need more information please let me know.

**Comment by lukeyeager** [comment]
As I said, 

> please ask your question at our digits-users google group
> (If you really want to know in detail on above ERROR sitation, please let me know.)
> Of course I do, I'd just prefer to do it elsewhere

https://groups.google.com/d/forum/digits-users

**Comment by KangKyungSoo** [comment]
OK, I will do it elsewhere(https://groups.google.com/d/forum/digits-users), 
after the test results of the caffe, for the reference.

**Comment by KangKyungSoo** [comment]
mv snapshot_iter_1601460.caffemodel snapshot_iter_1601460.caffemodel.org
mv bvlc_googlenet.caffemodel snapshot_iter_1601460.caffemodel

I replaced my trained model(snapshot_iter_1601460.caffemodel) in DIGITS with the downloaded model(bvlc_googlenet.caffemodel) from caffe as above commends. And, I got correct result as following image. 

![googlenet test](https://cloud.githubusercontent.com/assets/11133669/7254753/f3cb2c70-e87f-11e4-9534-0cb4f4274d71.png)

The ERROR seems to be in standard GoogLeNet traing model. Have you tested it?

**Comment by lukeyeager** [comment]
Fine, I give up. We can discuss this here.

For one thing, that reference model was trained differently. If you read [the details that caffe provides](https://github.com/BVLC/caffe/blob/c6414ea7cab5917c904729c87f786e0a5909475c/models/bvlc_googlenet/readme.md), you see that they are training for 60 epochs with a different learning rate setup. So if you leave all of the defaults in DIGITS, you're not going to get a model that works as well.

However, that doesn't explain why you're getting 0.1% confidence for all categories after 30 epochs - there's something else going on. Will you please post your network and solver like @joyofdata did in #84?

> The ERROR seems to be in standard GoogLeNet traing model. Have you tested it?

Of course I have. It works for me on smaller datasets, but I haven't run it on the full ImageNet dataset.

**Comment by KangKyungSoo** [comment]
Thank you.

solver.prototxt  : http://pastebin.com/9qNqjYf7
train_val.prototxt  : http://pastebin.com/TSinsCFP
deploy.prototxt : http://pastebin.com/3YRhyBc0
caffe_output.log : http://pastebin.com/gLPsc1VH

**Comment by lukeyeager** [comment]
One thing to note is that your learning rate policy is probably not doing what you're expecting. Your policy is "Exponential" and your "gamma" is `0.999996797097` (where are you getting those parameters from?), but that needs to be rescaled from the original max_iter to 100 since DIGITS sets the learning rate according to the percentage of the job done. 

    new_gamma = old_gamma * 100 / previous_max_iter

That's not very well documented yet - I need to make this more well known.

**Comment by lukeyeager** [comment]
Oddly, your first classifier is doing just fine:

    Test net output #1: loss1/top-1 = 0.465354
    Test net output #2: loss1/top-5 = 0.72203

But your last classifier is terrible:

    Test net output #7: loss3/top-1 = 0.00149632
    Test net output #8: loss3/top-5 = 0.00540976

I'm not sure what that means, exactly. You're clearly not using **all** of the DIGITS defaults because your learning rate policy is "Exponential" instead of "Step". What else did you change?

**Comment by KangKyungSoo** [comment]
I did not chage the default settings except for the learning rate policy, form 'Step Down' to 'Exponential', since it gave me higher training rate at AlexNet case.

**Comment by lukeyeager** [comment]
Ok, I'll try to find some time to do a GoogLeNet run on the full ImageNet dataset and get back to you.

**Comment by KangKyungSoo** [comment]
I really appreciate it.

**Comment by lukeyeager** [comment]
Ok, I'm seeing the same problem you are - the first classifier is making progress, but the second and third aren't.

![googlenet on imagenet screenshot](https://cloud.githubusercontent.com/assets/687269/7324047/cfc9674e-ea67-11e4-8df2-11c4d3f696ab.jpg)

I don't know why that's happening yet, but I can verify this as an issue. I'll keep at it, but debugging this stuff is slow since it takes so long to train.

**Comment by lukeyeager** [comment]
Ok, I tried two things and one of them worked:

1. I changed the batch size from 24 (DIGITS default) to 32 (caffe suggestion). That didn't work. I got the same graph as before (classifers 2 and 3 aren't moving).
![googlenet on imagenet screenshot 2](https://cloud.githubusercontent.com/assets/687269/7356318/268c7f00-ecdc-11e4-86bb-399d71b2131d.jpg)

2. I changed the learning rate from 0.01 (DIGITS default and caffe suggestion) to 0.005 (gut feeling). That did the trick.

![googlenet on imagenet screenshot 3](https://cloud.githubusercontent.com/assets/687269/7356328/36fe4404-ecdc-11e4-9d63-e58dd16a2622.jpg)

Will you try lowering your initial learning rate and see if that fixes the issue for you? I'm not content with this solution yet, but it's a step in the right direction.

**Comment by KangKyungSoo** [comment]
I will try, too.

**Comment by KangKyungSoo** [comment]
I got correct result of GoogLeNet, and it worked well.

Thank lukeyeager. 

![googlenet 2](https://cloud.githubusercontent.com/assets/11133669/7464893/6250b3cc-f302-11e4-8f4a-8cbf26610d84.png)

**Comment by lukeyeager** [comment]
> I got correct result of GoogLeNet, and it worked well.

Glad you got your issue worked out. I may need to allow different models to have different default learning rates or something. I'll leave this issue open for now to remind me to fix that.

**Comment by catsdogone** [comment]
I got correct result of GoogLeNet, but the accuracy is not high.
<img width="531" alt="default" src="https://cloud.githubusercontent.com/assets/7392509/10126503/58ba1a74-65c6-11e5-8e57-10a0284aea02.PNG">
  
My training data:
<img width="248" alt="buhuo" src="https://cloud.githubusercontent.com/assets/7392509/10126764/97e1f77c-65ca-11e5-8795-5da5bbdb1d95.PNG">

My network and solver
<img width="669" alt="45" src="https://cloud.githubusercontent.com/assets/7392509/10127013/49ee972e-65ce-11e5-9355-c3fab17db36a.PNG">

**Comment by KangKyungSoo** [comment]
Image data set : SQUASH give us more accuracy than CROP.

**Comment by catsdogone** [comment]
@KangKyungSoo 
Thanks very much! I will try.
